### PR TITLE
React-basic 15 updates

### PR DESCRIPTION
**Pull request by spicydonuts**
In `react-basic` 15, I split out the `React.Basic.DOM` modules into their own package, `react-basic-dom`. Since most users will probably be choosing `react-native-basic` or `react-basic-dom` instead of using both, I copied the one bit of the DOM modules that were being used here (CSS) directly into the repo.

`react-basic` 15 also removes the component implementation that lived in the `React.Basic` module. It's now in `React.Basic.Classic`, in its own `react-basic-classic` package. This was only relevant in the examples.

### CODE REVIEW

File: packages.dhall [hunk 1]
```diff
@@ -1,7 +1,7 @@
 let upstream =
       https://github.com/purescript/package-sets/releases/download/psc-0.13.5-20200103/packages.dhall sha256:0a6051982fb4eedb72fbe5ca4282259719b7b9b525a4dda60367f98079132f30
 
-let overrides = {=}
+let overrides = { react-basic = upstream.react-basic // { version = "v15.0.0", dependencies = [ "effect", "record" ] } }
```
Comment (spicydonuts) [inline packages.dhall:4]: I don't know what the correct ordering is here, whether this package should temporarily contain an override or whether this PR should wait for an updated packages set and remove these overrides/additions (additions in the example `packages.dhall` files). The package set PR is [here](https://github.com/purescript/package-sets/pull/652/files#diff-1ef6b8e409a01edc78fb84eb3d4489b3R7). This package's filename and repo probably need to be moved/updated anyway though, and the package set does build as-is, even if the dom dependency isn't ideal, so it's probably fine to wait for the package set update.

Comment (spicydonuts) [reply packages.dhall]: Updated to the latest package-set and removed the overrides

**Comment by f-f** [comment]
@spicydonuts for some weird reason I never got the notification for this PR, I'm so sorry! :see_no_evil: 

I will merge now and I added you as a collaborator so you'll be able to move things along if I'm unavailable :blush:

**Comment by spicydonuts** [comment]
no worries :)

**Pull request merged**